### PR TITLE
Add runtime config parameter to force ijwhost to load assemblies in an isolated context

### LINKED ISSUE #104480: Any way to revert to pre-7.0 isolated assembly loading in C++/CLI?
I have a framework using C++/CLI to load managed plugins within a native application.

This change to loading assemblies in the default ALC:

 https://github.com/dotnet/runtime/pull/66486

has broken my ability to load multiple plugins.

Is there any way to revert to the previous isolated assembly loading behavior?

### PR DESCRIPTION
**Pull request by mikeoliphant**
This PR adds a runtime config parameter ("System.Runtime.InteropServices.CppCLI.LoadComponentInIsolatedContext") to indicated that ijwhost should load assemblies in an isolated load context.

This is done via a change to InMemoryAssemblyLoader.LoadInMemoryAssemblyInContext to allow specifying either default load context or isolated load context.

The PR also adds ijwhost tests for use of the new config parameter.

Resolves https://github.com/dotnet/runtime/issues/104480

### CODE REVIEW

File: src/installer/tests/HostActivation.Tests/NativeHosting/Ijwhost.cs [hunk 1]
```diff
@@ -53,6 +54,46 @@ public void LoadLibrary(bool no_runtimeconfig)
...
+                    app.AppDll,
+                    "NativeEntryPoint"
+                };
+
+                if (load_isolated)
+                {
+                    RuntimeConfig.FromFile(app.RuntimeConfigJson)
+                    .WithProperty("System.Runtime.InteropServices.CppCLI.LoadComponentInIsolatedContext", load_isolated.ToString())
+                    .Save();
+                }
```
[the remark this answers is not in the document]
Comment (mikeoliphant) [reply src/installer/tests/HostActivation.Tests/NativeHosting/Ijwhost.cs]: Yep. Fixed with latest commit.

**Review by elinor-fung — APPROVED** [review]
Thank you!

**Comment by mikeoliphant** [comment]
👍 

I know it is probably hard to say definitively, but any idea when this change might see an official release? Thanks.

**Comment by mikeoliphant** [comment]
> The main branch is for .NET 9 RC1 now, which would be early September. I'll also see if we can get this approved to backport to .NET 8 - that would also be September.

Perfect - thanks!

**Pull request merged**